Api/src/guild-module-qotd/guild-module-qotd.service.ts

```diff
@@ -26,11 +32,35 @@ export class GuildModuleQotdService {
     });
   }
 
-  update(id: number, updateGuildModuleQotdInput: UpdateGuildModuleQotdInput) {
-    return `This action updates a #${id} guildModuleQotd`;
+  getGuild(id: number): Promise<Guild> {
+    return this.guildService.findOne(id);
   }
 
-  remove(id: number) {
-    return `This action removes a #${id} guildModuleQotd`;
+  async update(
+    id: number,
+    updateGuildModuleQotdInput: UpdateGuildModuleQotdInput,
+  ): Promise<GuildModuleQotd> | null {
+    const guildModuleQotd: GuildModuleQotd =
+      await this.guildModuleQotdRepository.findOne({
+        where: { guild_id: id },
+      });
+    if (guildModuleQotd) {
+      return this.guildModuleQotdRepository.save({
+        ...guildModuleQotd,
+        ...updateGuildModuleQotdInput,
+      });
+    }
+    throw new Error('GuildModuleQotd not found');
+  }
+
+  async remove(id: number): Promise<GuildModuleQotd> | null {
```

Comment:
Same again, check if null is a valid return type, I could be wrong, just don't remember it being one.